specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,400 @@
+
+```cpp
+int main()
+{
+    PushActivatorDetails details(winrt::guid("BACCFA91-F1DE-4CA2-B80E-90BE66934EC6"));
+
+    // Registers a Push Trigger and sets up an inproc COM Server for Activations
+    PushManager::RegisterActivator(details, PushRegistrationType::PushTrigger | PushRegistrationType::ComActivator);
+
+    // Check to see if the WinMain activation is due to a Push Activator
+    // Note: This API is currently not present in Reunion but will be included as part of the AppLifecycle Public API spec.
+    auto activation = AppLifecycle::Activation();
+    auto kind = activation.Kind();
+
+    if (kind == ReunionActivationKind::Push)
+    {
+        auto pushArgs = activation.Arguments().try_as<PushReceivedEventArgs>();
+
+        // Call TakeDeferral to ensure that code runs in low power
+        pushArgs.GetDeferral();
+
+        auto payload = pushArgs.Payload();
+
+        // Do stuff to process the raw payload
+
+        // Call CompleteDeferral as good practise: Needed mainly for low power usage
+        pushArgs.CompleteDeferral();
+    }
+    else if (kind == ReunionActivationKind::Launch) // This indicates that the app is launching in the foreground
+    {
+        // Register an event to Intercept Push payloads
+        auto eventToken = details.PushReceived([](const auto&, PushReceivedEventArgs args)
+        {
+                // Call TakeDeferral to ensure that code runs in low power
+                args.GetDeferral();
+
+                auto payload = args.Payload();
+
+                // Do stuff to process the raw payload
+
+                // Call CompleteDeferral as good practise: Needed mainly for low power usage
+                args.CompleteDeferral();
+        });
+
+        auto channelOperation = PushManager::CreateChannelAsync(winrt::guid("F80E541E-3606-48FB-935E-118A3C5F41F4"));
+
+        // Setup the inprogress event handler
+        channelOperation.Progress(
+            [](
+                IAsyncOperationWithProgress<ChannelResult, ChannelResult> const& /* sender */,
+                ChannelResult const& args)
+            {
+                if (args.Status() == ChannelStatus::InProgress)
+                {
+                    // This is basically a noop since it isn't really an error state
+                    printf("The first channel request is still in progress! \n");
+                }
+                else if (args.Status() == ChannelStatus::InProgressRetry)
+                {
+                    LOG_HR_MSG(args.ExtendedError(), "The channel request is in back-off retry mode because of a retryable error! Expect delays in acquiring it.");
+                }
+            });
+
+        // Setup the completed event handler
+        channelOperation.Completed(
+            [](
+                IAsyncOperationWithProgress<ChannelResult, ChannelResult> const& sender,
+                AsyncStatus const /* asyncStatus */)
+            {
+                auto result = sender.GetResults();
```

Comment:
This will throw if the operation encountered a COM failure, e.g. E_CANCELLED.

Reply:
I added a check to look at the AsyncStatus parameter to see if we have Completed.